android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -195,7 +206,12 @@ constructor(
             source.setParameter(Task.SP_PERIOD, period)
             source.setParameter(ActivityDefinition.SP_VERSION, IntegerType(index))
 
-            val structureMap = fhirEngine.get<StructureMap>(IdType(action.transform).idPart)
+            val structureMap =
+              if (appIdExistsAndIsNotNull(sharedPreferencesHelper)) {
+                configurationRegistry.retrieveResourceFromConfigMap<StructureMap>(
+                  resourceId = IdType(action.transform).idPart,
+                )
+              } else fhirEngine.get<StructureMap>(IdType(action.transform).idPart)
```

Comment:
Updating this would not be needed. If the retrieved configs were added to the database they can be read from the DB directly instead of loading them in memory.